src/phpDocumentor/Descriptor/DocumentationSetDescriptor.php

```diff
@@ -13,12 +13,17 @@
 
 namespace phpDocumentor\Descriptor;
 
+use phpDocumentor\Dsn;
+
 abstract class DocumentationSetDescriptor
 {
     /** @var string */
     protected $name = '';
 
-    /** @var array */
+    /**
+     * @phpstan-var array{dsn?: Dsn, paths?: array<int, string>}
```

Comment:
@jaapio Do you see this? PHPStan actually implemented a variant of the associative array syntax that we proposed in PSR-5 :) Perhaps worth checking out the adoption rate of this

Reply:
Yes I know the existence of this notations. Although I don't like the notation it makes a lot of sense for legacy projects.
I do agree that we need to adopt this, all static analyzers (3) in php do support this.
We might want to check this with the guys from jetbrains. They are in the psr-5 group

Reply:
Note that PHPStan and Psalm try to have similar syntaxes (both reads each other @xxx- notations while scanning).
Here are the notations supported by psalm: https://psalm.dev/docs/annotating_code/type_syntax/array_types/